fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -233,69 +237,60 @@ protected RdfStream getResourceTriples(final int limit) {
 
         final LdpPreferTag ldpPreferences = new LdpPreferTag(returnPreference);
 
-        final RdfStream rdfStream = new RdfStream();
-
         final Predicate<Triple> tripleFilter = ldpPreferences.prefersServerManaged() ? x -> true :
             IS_MANAGED_TRIPLE.negate();
 
-        if (ldpPreferences.prefersServerManaged()) {
-            rdfStream.concat(getTriples(LdpRdfContext.class));
-        }
-
-        rdfStream.concat(filter(getTriples(TypeRdfContext.class), tripleFilter::test));
+        final List<Stream<Triple>> streams = new ArrayList<>();
 
-        rdfStream.concat(filter(getTriples(PropertiesRdfContext.class), tripleFilter::test));
+        streams.add(getTriples(PROPERTIES).filter(tripleFilter));
 
-        if (!returnPreference.getValue().equals("minimal")) {
+        if (returnPreference.getValue().equals("minimal")) {
+            if (ldpPreferences.prefersServerManaged()) {
+                streams.add(getTriples(SERVER_MANAGED)
```

Comment:
@awoods: that particular concern is a non-issue here. The `FedoraResource::getTriples` method accepts an `EnumSet`, so if a particular class of triples is to be `include`d or `omit`ed, the implementing client either includes or omits that particular `RdfContext` value: `obj.getTriples(translator(), of(PROPERTIES, SERVER_MANAGED))` or `obj.getTriples(translator(), of(LDP_MEMBERSHIP))`.

The bigger issue is what @ajs6f mentions: namely that the `enum` can't be extended by any sort of hierarchical mechanism that way a class or interface can. 

What I would like to resolve is whether that change (from an `enum` to an `interface`) needs to happen now, as part of this PR, or later, after this gets merged. My opinion is that it should happen later, but only because rebasing this PR is truly hellish, and if there are too many more commits that get merged in the meantime, it will become unbearable.